source/elements/oneMKL/source/domains/blas/dgmm_batch.rst

```diff
@@ -0,0 +1,499 @@
+      entries must be positive and ``ldc[i]`` must be at least
+      ``m[i]`` if column major layout is used to store matrices or at
+      least ``n[i]`` if row major layout is used to store matrices.
+
+   group_count
+      Specifies the number of groups. Must be at least 0.
+
+   group_size
+      Array of ``group_count`` integers. ``group_size[i]`` specifies the
+      number of diagonal matrix-matrix product operations in group ``i``.
+      All entries must be at least 0.
+
+   dependencies
+         List of events to wait for before starting computation, if any.
+         If omitted, defaults to no dependencies.
+
+.. container:: section
+
+   .. rubric:: Output Parameters
+
+   c
+      Output overwritten by ``batch_size`` diagonal matrix-matrix product
+      operations.
+
+.. container:: section
+
+   .. rubric:: Return Values
+
+   Output event to wait on to ensure computation is complete.
+
+**Strided API**
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::column_major {
+       sycl::event dgmm_batch(sycl::queue &queue,
+                              onemkl::mkl::side left_right,
+                              std::int64_t m,
+                              std::int64_t n,
+                              const T *a,
+                              std::int64_t lda,
+                              std::int64_t stridea,
+                              const T *b,
+                              std::int64_t incx,
+                              std::int64_t stridex,
+                              T *c,
+                              std::int64_t ldc,
+                              std::int64_t stridec,
+                              std::int64_t batch_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::row_major {
+       sycl::event dgmm_batch(sycl::queue &queue,
+                              onemkl::mkl::side left_right,
+                              std::int64_t m,
+                              std::int64_t n,
+                              const T *a,
+                              std::int64_t lda,
+                              std::int64_t stridea,
+                              const T *b,
+                              std::int64_t incx,
+                              std::int64_t stridex,
+                              T *c,
+                              std::int64_t ldc,
+                              std::int64_t stridec,
+                              std::int64_t batch_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   queue
+      The queue where the routine should be executed.
+
+   left_right
+      Specifies the position of the diagonal matrix in the product.
+      See :ref:`onemkl_datatypes` for more details.
+
+   m
+      Number of rows of op(``A``). Must be at least zero.
```

Comment:
no op for A. we dont have transpose A paramater.

Reply:
adjusted